src/command_line.cc

```diff
@@ -1211,7 +1211,9 @@ WorkThread::Result IndexMain(Config* config,
                              WorkingFiles* working_files,
                              MultiQueueWaiter* waiter,
                              QueueManager* queue) {
-  EmitProgress(queue);
+  if (config->enableProgressReports) {
+    EmitProgress(queue);
+  }
```

Comment:
So from an API point of view, it'd be nice to have the config switch sit inside the method itself so that EmitProgress just silently no-ops, but there's no reference chain back to the config from QueueManager.

Reply:
EmitProgress should take a Config instance as a parameter and do this check internally.

Reply:
Wilco, I'll update.